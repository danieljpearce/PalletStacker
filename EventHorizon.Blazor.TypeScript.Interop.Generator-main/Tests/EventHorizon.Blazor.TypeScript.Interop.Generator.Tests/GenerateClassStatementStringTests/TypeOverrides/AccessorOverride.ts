declare module Examples {
    export class ExampleClass {
        static get vectorStaticCheck(): Vector3;
        static get primativeStaticCheck(): number;

        get vectorCheck(): Vector3;
        get primativeCheck(): number;
    }
}