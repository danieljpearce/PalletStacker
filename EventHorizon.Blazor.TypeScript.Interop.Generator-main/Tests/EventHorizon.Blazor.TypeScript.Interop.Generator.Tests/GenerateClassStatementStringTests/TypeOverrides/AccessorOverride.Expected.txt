/// Generated - Do Not Edit
namespace Examples
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using EventHorizon.Blazor.Interop.Callbacks;
    using EventHorizon.Blazor.Interop.ResultCallbacks;
    using Microsoft.JSInterop;

    
    
    [JsonConverter(typeof(CachedEntityConverter<ExampleClass>))]
    public class ExampleClass : CachedEntityObject
    {
        #region Static Accessors
        private static OverrideVector4 __vectorStaticCheck;
        public static OverrideVector4 vectorStaticCheck
        {
            get
            {
            if(__vectorStaticCheck == null)
            {
                __vectorStaticCheck = EventHorizonBlazorInterop.GetClass<OverrideVector4>(
                    "Examples",
                    "ExampleClass.vectorStaticCheck",
                    (entity) =>
                    {
                        return new OverrideVector4() { ___guid = entity.___guid };
                    }
                );
            }
            return __vectorStaticCheck;
            }
        }

        
        public static int primativeStaticCheck
        {
            get
            {
            return EventHorizonBlazorInterop.Get<int>(
                    "Examples",
                    "ExampleClass.primativeStaticCheck"
                );
            }
        }
        #endregion

        #region Static Properties

        #endregion

        #region Static Methods

        #endregion

        #region Accessors
        private OverrideVector4 __vectorCheck;
        public OverrideVector4 vectorCheck
        {
            get
            {
            if(__vectorCheck == null)
            {
                __vectorCheck = EventHorizonBlazorInterop.GetClass<OverrideVector4>(
                    this.___guid,
                    "vectorCheck",
                    (entity) =>
                    {
                        return new OverrideVector4() { ___guid = entity.___guid };
                    }
                );
            }
            return __vectorCheck;
            }
        }

        
        public int primativeCheck
        {
            get
            {
            return EventHorizonBlazorInterop.Get<int>(
                    this.___guid,
                    "primativeCheck"
                );
            }
        }
        #endregion

        #region Properties

        #endregion
        
        #region Constructor
        public ExampleClass() : base() { }

        public ExampleClass(
            ICachedEntity entity
        ) : base(entity)
        {
            ___guid = entity.___guid;
        }


        #endregion

        #region Methods

        #endregion
    }
}